subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -226,7 +226,12 @@ static void init_rest_client_request(struct nrf_cloud_rest_context const *const
 	req->port		= HTTPS_PORT;
 	req->host		= CONFIG_NRF_CLOUD_REST_HOST_NAME;
 	req->tls_peer_verify	= TLS_PEER_VERIFY_REQUIRED;
-	req->timeout_ms		= rest_ctx->timeout_ms;
+
+	if (rest_ctx->timeout_ms <= 0) {
```

Comment:
Probably should check just ==0 here. Negative values are reserved for special values like `SYS_FOREVER_MS`.

Right now `SYS_FOREVER_MS` is the only such special value, and I don't foresee more being added, but I still think it is better to exactly match our informal special value of zero

Reply:
i was trying to catch undefined values here... so if -2 is used, it isn't left up to some other library to handle it.

Reply:
First off, doing a bit more digging, only specifically `-1` is reserved. All other non-reserved negative integers are converted to `0`. So I was wrong in my original statement.

Definitely do update the documentation, though!

Or consider not making **_any_** values synonymous with `SYS_FOREVER_MS`, but that recommendation might be overly extreme. I'm really on the fence surrounding this topic